larsim/PhotonPropagation/PhotonLibrary.h

```diff
@@ -65,7 +72,26 @@ namespace phot{
     virtual int NVoxels() const override { return fNVoxels; }
 
     virtual bool isVoxelValid(size_t Voxel) const override { return isVoxelValidImpl(Voxel); }
-
+    
+    
+    // --- BEGIN --- Metadata: voxel information -------------------------------
+    /// @name Metadata: voxel information
+    /// @{
+    
+    /// Returns whether voxel metadata is available.
+    bool hasVoxelDef() const { return fVoxelDef.has_value(); }
+
+    /// Returns the current voxel metadata (undefined behaviour if none).
+    /// @see `hasVoxelDef()`
+    sim::PhotonVoxelDef const& GetVoxelDef() const { return fVoxelDef.value(); }
```

Comment:
Actually the behavior is well-defined in this case: an exception will be thrown if the optional object contains no value.  Perhaps you'd like to do the following instead?
```.cpp
sim::PhotonVoxelDef const& GetVoxelDef() const noexcept 
{
  assert(hasVoxelDef());
  return *fVoxelDef;
}
```

Reply:
Yes, the behaviour of the current implementation is predictable, but I don't want to create expectations.
I accept the suggested implementation.

Reply:
Commit c497e35.